devito/builtins.py

```diff
@@ -283,16 +306,10 @@ def initialize_function(function, data, nbl, mapper=None, mode='constant',
             function.data[:] = data[:]
         return
 
-    if len(as_tuple(nbl)) == 1 and len(as_tuple(nbl)) < function.ndim:
-        nbl = function.ndim*(as_tuple(nbl)[0], )
-    elif len(as_tuple(nbl)) == function.ndim:
-        pass
-    else:
-        raise ValueError("nbl must be an integer or tuple of integers of length" +
-                         " function.shape.")
+    nbl = nbl_to_padsize(nbl, function.ndim)
 
-    slices = tuple([slice(n, -n) for _, n in zip(range(function.grid.dim),
-                                                 as_tuple(nbl))])
+    slices = tuple([slice(n[0], -n[1]) for _, n in zip(range(function.grid.dim),
```

Comment:
not a big fan of all these n[0] and n[1] ... we can instead make it implicit in the iteration, which has the added value of implicitly checking the size of the tuple 

```
=  tuple([slice(n0, -n1) for _, (n0, n1) in zip(range(function.grid.dim),
```

or alternatively here  you could have just unpacked `n` with `*n`

Reply:
If going down this route I'd prefer `nl`/`nleft` and `nr`/`nright` over `n0`, `n1`.

Reply:
done